### PR TITLE
feat(condo): DOMA-7554 meter search service for residents

### PR DESCRIPTION
**Pull request by sitozzz**
This custom service provide an ability for residents to find all meters related by address and accountNumber.
It became necessary when organizations with resource provider type will be able to create own meters on same address.
This special case will be handled by `MeterResourceOwner` records

### CODE REVIEW

File: apps/condo/domains/meter/access/AllResidentMetersService.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * Generated by `createservice meter.AllResidentMetersService`
+ */
+const { isEmpty } = require('lodash')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { find } = require('@open-condo/keystone/schema')
+
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+
+async function canAllResidentMeters ({ authentication: { item: user }, args: { where: { id: residentId } } }) {
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/access/AllResidentMetersService.js:11]: `canExecute...`

File: apps/condo/domains/meter/gql.js [hunk 2]
```diff
@@ -52,6 +52,12 @@ const DELETE_METER_AND_METER_READINGS_MUTATION = gql`
 const METER_RESOURCE_OWNER_FIELDS = `{ organization { id } resource { id } address addressKey ${COMMON_FIELDS} }`
 const MeterResourceOwner = generateGqlQueries('MeterResourceOwner', METER_RESOURCE_OWNER_FIELDS)
 
+const ALL_RESIDENT_METERS_QUERY = gql`
+    query allResidentMeters ($where: ResidentWhereUniqueInput!, $first: Int, $skip: Int) {
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/gql.js:56]: Maybe call first argument resident for all mobile users? and N optional args for query-specific logic

File: apps/condo/domains/meter/schema/AllResidentMetersService.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                }
+
+                const userConsumers = await find('ServiceConsumer', {
+                    resident: { id: residentId, deletedAt: null },
+                    organization: { deletedAt: null },
+                    deletedAt: null,
+                })
+
+                const accountNumberWhere = {
+                    OR: userConsumers.map(serviceConsumer => ({ AND: [{ accountNumber: serviceConsumer.accountNumber }] })),
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/schema/AllResidentMetersService.js:82]: `accountNumber_in`

File: apps/condo/domains/meter/schema/AllResidentMetersService.test.js [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+
+            const [originalBillingAccount] = await createTestBillingAccount(admin, originalContext, originalBillingProperty)
+            const [serviceProviderBillingAccount] = await createTestBillingAccount(admin, serviceProviderContext, serviceProviderBillingProperty)
+
+            const unitName = faker.random.alphaNumeric(8)
+            const unitType = FLAT_UNIT_TYPE
+
+            const [resident] = await createTestResident(admin, residentClient.user, originalProperty, { unitName, unitType })
+            await createTestServiceConsumer(admin, resident, originalOrganization, { accountNumber: originalBillingAccount.number })
+            // const [serviceProviderResident] = await createTestResident(admin, residentClient.user, serviceProviderProperty, { unitName, unitType })
```
Comment (SavelevMatthew) [inline apps/condo/domains/meter/schema/AllResidentMetersService.test.js:166]: remove comment

**Comment by sitozzz** [comment]
I'm closing this because I found a more accurate solution https://github.com/open-condo-software/condo/pull/4123

**Pull request closed without merging**